src/ripple/protocol/impl/Feature.cpp

```diff
@@ -20,17 +20,43 @@
 #include <BeastConfig.h>
 #include <ripple/protocol/digest.h>
 #include <ripple/protocol/Feature.h>
+
+#include <boost/container/flat_map.hpp>
+
 #include <cstring>
 
 namespace ripple {
 
+static boost::container::flat_map<uint256, std::size_t> featureToIndex;
+static std::vector<uint256> features;
```

Comment:
Yeah, I'm less concerned about the immediate need for thread safety.  It's more programming in the future tense.  There's no reason that `FeatureBitset` must be limited to Env use.  If it starts getting more widespread use I'd prefer not to have it cause mysterious crashes or, worse, odd non-crashing behavior.  My preference would be to build in the thread safety now.